### PR TITLE
Edit CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by AnotherButler**
Make grammar edits to CONTRIBUTING.md

@adbridge @mazimkhan @bridadan Some of this content seems to be repeated. Would you recommend rearranging or deleting any of it?

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,86 +1,81 @@
...
-Please report all bugs using the Issues tab on the GitHub web-page. It will help us to collaborate on issues more promptly. One of our gate-keepers (developers responsible for quality and the repository) will review the issue and assign a label such as _bug_, _enhancement_, _help wanted_ or _wontfix_.
-# How to contribute
-You can either file a bug, help fix a bug or propose a new feature (or enhancement) and implement it yourself. If you want to contribute please:
-* Bug reports: File a bug report in the Issues tab of this repo to let us know there are problems with our code.
+
+Please report all bugs using the Issues tab on the GitHub page. This process helps us collaborate on issues promptly. One of our gate-keepers (developers responsible for quality and the repository) will review the issue and assign a label, such as _bug_, _enhancement_, _help wanted_ or _wontfix_.
+
+## How to contribute
+
+You can file a bug, help fix a bug or propose a new feature (or enhancement) and implement it yourself. If you want to contribute, please:
```
Comment (bridadan) [inline CONTRIBUTING.md]: I think this sentence ends a bit abruptly. Maybe say something like `If you want to contribute, please see below?` What do you think @AnotherButler?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,86 +1,81 @@
...
+
+* Bug reports: File a bug report in the Issues tab of this repo to let us know there are problems with the code.
   * Make sure your bug report contains a simple description of the problem.
   * Include information about the host computer's configuration and OS or VM used.
-  * Include information about the application's version. All applications should have at least a ``--version`` switch you can use to check the version.
-  * Copy/paste useful console dumps and configuration files' content. Please use [fenced code blocks](https://help.github.com/articles/github-flavored-markdown/#fenced-code-blocks) to encapsulate code snippets.
+  * Include information about the application's version. All applications should have at least a `--version` switch you can use to check the version.
+  * Copy and paste useful console dumps and configuration files' content. Please use [fenced code blocks](https://help.github.com/articles/github-flavored-markdown/#fenced-code-blocks) to encapsulate code snippets.
```
Comment (bridadan) [inline CONTRIBUTING.md]: Looks like the link is dead, should be this one here: `https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -97,44 +92,52 @@ $ git push origin issue_38
...
 
-### Code Like a Pythonista: Idiomatic Python
-Please do your best to follow [Idiomatic Python](http://python.net/~goodger/projects/pycon/2007/idiomatic/handout.html) interactive tutorial.
+This chapter attempts to explain the basic styles and patterns in mbed test tools projects. Do the following to produce new code and update existing code.
+
+### Code like a Pythonista: idiomatic Python
+
+Please do your best to follow the [Idiomatic Python](http://python.net/~goodger/projects/pycon/2007/idiomatic/handout.html) interactive tutorial.
+
+### Coding style: readability matters
```
Comment (bridadan) [inline CONTRIBUTING.md]: At this point, seems like this section is really necessary, perhaps we could remove it?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -147,57 +150,78 @@ def make_squares(key, value=0):
...
 $ git fetch upstream
 $ git rebase upstream/master
 ```
-The goal of the rebase is to have a cleaner history if I have local changes or commits on the repo.
-# Final notes
-1. Please do not change the version of the package in the ``setup.py`` file. The person or process responsible for releasing will do this and release the new version.
-2. Keep your GitHub updated! Please make sure you are rebasing your local branch with changes so you are up to date and it is possible to automatically merge your pull request.
-3. Please, if possible, squash your commits before pushing to the remote repository.
+
+Rebasing creates a cleaner history for those with local changes or commits on the repo.
```
Comment (bridadan) [inline CONTRIBUTING.md]: I'd recommend changing this line to `Rebasing creates a cleaner history for developers with local changes or commits.`

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -147,57 +150,78 @@ def make_squares(key, value=0):
...
+
+## Final notes
+
+1. Please do not change the version of the package in the `setup.py` file. The person or process responsible for releasing will do this and release the new version.
+2. Keep your GitHub updated! Please make sure you are rebasing your local branch with changes, so it up to date and we can automatically merge your pull request.
```
Comment (bridadan) [inline CONTRIBUTING.md]: I think this line should be changed to the following:

```
Keep your GitHub fork updated! Please make sure you are rebasing your local branch with changes, so it is up to date and we can automatically merge your pull request.
```

**Comment by AnotherButler** [comment]
Thanks for reviewing. I've made all of your suggested changes.

**Comment by bridadan** [comment]
Ready for your eyes and/or merge @adbridge

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -215,13 +208,13 @@ $ git fetch upstream
...
 ```
 
-Rebasing creates a cleaner history for those with local changes or commits on the repo.
+Rebasing creates a cleaner history for developers with local changes or commits.
 
 ## Final notes
 
 1. Please do not change the version of the package in the `setup.py` file. The person or process responsible for releasing will do this and release the new version.
-2. Keep your GitHub updated! Please make sure you are rebasing your local branch with changes, so it up to date and we can automatically merge your pull request.
+2. Keep your GitHub fork updated! Please make sure you are rebasing your local branch with changes, so it up to date and we can automatically merge your pull request.
```
Comment (adbridge) [inline CONTRIBUTING.md]: ..so it is up to date...

**Comment by AnotherButler** [comment]
Great eye. Thanks, @adbridge. I have fixed this.

**Pull request merged**